### PR TITLE
stableAsset precompile fixes

### PR DESCRIPTION
**Pull request by ermalkaleci**

### CODE REVIEW

File: runtime/common/src/precompile/stable_asset.rs [hunk 1]
```diff
@@ -394,6 +421,18 @@ mod tests {
 			"};
 			assert_eq!(resp.exit_status, ExitSucceed::Returned);
 			assert_eq!(resp.output, expected_output.to_vec());
+
+			// empty output if pool doesn't exists
+
+			// getStableAssetPoolTokens(uint32) -> 0x7172c6aa
```
Comment (ferrell-code) [inline runtime/common/src/precompile/stable_asset.rs:427]: comment has different function selector

File: runtime/common/src/precompile/stable_asset.rs [hunk 2]
```diff
@@ -86,115 +86,140 @@ where
...
+							.map(|x| InputPricer::<Runtime>::read_currency(*x))
+							.sum::<u64>(),
+					);
+					// make sure there's enough gas
+					if let Some(gas_limit) = target_gas {
+						if gas_limit < gas_cost {
+							return Err(PrecompileFailure::Error {
+								exit_status: ExitError::OutOfGas,
+							});
 						}
```
Comment (ferrell-code) [inline runtime/common/src/precompile/stable_asset.rs:105]: Could move this logic to `Pricer`, I think overlay would make it take the same amount of time. This way is fine too

**Review by ferrell-code — APPROVED** [review]

File: runtime/common/src/precompile/stable_asset.rs (hunk 2 above)
Comment (ferrell-code) [reply runtime/common/src/precompile/stable_asset.rs:105]: Yeah make sense, would iterate over twice

**Review by xlc — APPROVED** [review]
make sure we have docs on the contract side indicates those behaviour

**Review by zjb0807 — APPROVED** [review]
LGTM. 
I think `output: Default::default()` is better. Can you update others in #2152?

**Comment by ermalkaleci** [comment]
> LGTM.
> I think `output: Default::default()` is better. Can you update others in #2152?

I will have a look. The problem is we can't update interfaces for without some kind of versioning

**Comment by ermalkaleci** [comment]
on the interface I will have `function getStableAssetPoolTotalSupply(uint32 poolId) external view returns (bool, uint256)` where bool indicates if pool exists

**Comment by xlc** [comment]
stable asset contract is not yet deployed to Karura / Acala so it is not too late to change it

**Pull request merged**